File: backend/src/main/java/com/dormitory/management/entity/Repair.java
package com.dormitory.management.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDateTime;

@Data
@EqualsAndHashCode(callSuper = false)
@TableName("repair")
public class Repair {

    @TableId(value = "id", type = IdType.ASSIGN_ID)
    private Long id;

    /**
     * 宿舍ID
     */
    @TableField("room_id")
    private Long roomId;

    /**
     * 宿舍号
     */
    @TableField("room_no")
    private String roomNo;

    /**
     * 楼栋名称
     */
    @TableField("building_name")
    private String buildingName;

    /**
     * 报修学生ID
     */
    @TableField("student_id")
    private Long studentId;

    /**
     * 报修学生姓名
     */
    @TableField("student_name")
    private String studentName;

    /**
     * 报修标题
     */
    @TableField("title")
    private String title;

    /**
     * 报修描述
     */
    @TableField("description")
    private String description;

    /**
     * 报修类型：water-水电，electric-电路，door-门窗，furniture-家具，other-其他
     */
    @TableField("type")
    private String type;

    /**
     * 优先级：1-低，2-中，3-高
     */
    @TableField("priority")
    private Integer priority;

    /**
     * 状态：1-待处理，2-处理中，3-已完成，4-已取消
     */
    @TableField("status")
    private Integer status;

    /**
     * 报修时间
     */
    @TableField("report_time")
    private LocalDateTime reportTime;

    /**
     * 处理时间
     */
    @TableField("handle_time")
    private LocalDateTime handleTime;

    /**
     * 完成时间
     */
    @TableField("complete_time")
    private LocalDateTime completeTime;

    /**
     * 处理人ID
     */
    @TableField("handler_id")
    private Long handlerId;

    /**
     * 处理人姓名
     */
    @TableField("handler_name")
    private String handlerName;

    /**
     * 处理备注
     */
    @TableField("handle_remark")
    private String handleRemark;

    /**
     * 报修图片，多个图片用逗号分隔
     */
    @TableField("images")
    private String images;

    /**
     * 联系电话
     */
    @TableField("contact_phone")
    private String contactPhone;

    /**
     * 创建时间
     */
    @TableField(value = "create_time", fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @TableField(value = "update_time", fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    /**
     * 创建人
     */
    @TableField(value = "create_by", fill = FieldFill.INSERT)
    private String createBy;

    /**
     * 更新人
     */
    @TableField(value = "update_by", fill = FieldFill.INSERT_UPDATE)
    private String updateBy;

    /**
     * 逻辑删除标记：0-未删除，1-已删除
     */
    @TableLogic
    @TableField("deleted")
    private Integer deleted;
}

File: frontend/src/api/repair.ts
import request from '@/utils/request'

// 报修信息接口定义
export interface Repair {
  id: number
  roomId: number
  roomNo: string
  buildingName: string
  studentId: number
  studentName: string
  title: string
  description: string
  type: string
  typeText: string
  priority: number
  priorityText: string
  status: number
  statusText: string
  reportTime: string
  handleTime: string | null
  completeTime: string | null
  handlerId: number | null
  handlerName: string | null
  handleRemark: string | null
  imageList: string[]
  contactPhone: string | null
  createTime: string
  updateTime: string
}

// 报修表单接口
export interface RepairForm {
  roomId: number
  studentId: number
  title: string
  description: string
  type: string
  priority: number
  contactPhone: string
  images: string[]
}

// 报修查询参数
export interface RepairParams {
  current: number
  size: number
  roomNo?: string
  roomId?: number
  buildingId?: number
  studentId?: number
  studentName?: string
  type?: string
  status?: number
  priority?: number
  startDate?: string
  endDate?: string
}

// 报修处理表单
export interface RepairHandleForm {
  id: number
  status: number
  handleRemark: string
}

/**
 * 分页查询报修列表
 */
export function getRepairPage(params: RepairParams) {
  return request({
    url: '/repair/page',
    method: 'post',
    data: params
  })
}

/**
 * 根据ID查询报修详情
 */
export function getRepairById(id: number): Promise<Repair> {
  return request({
    url: `/repair/${id}`,
    method: 'get'
  })
}

/**
 * 新增报修
 */
export function addRepair(data: RepairForm) {
  return request({
    url: '/repair',
    method: 'post',
    data
  })
}

/**
 * 处理报修
 */
export function handleRepair(data: RepairHandleForm) {
  return request({
    url: '/repair/handle',
    method: 'post',
    data
  })
}

/**
 * 删除报修
 */
export function deleteRepair(id: number) {
  return request({
    url: `/repair/${id}`,
    method: 'delete'
  })
}

/**
 * 获取宿舍选项列表
 */
export function getDormitoryOptions() {
  return request({
    url: '/repair/dormitories/options',
    method: 'get'
  })
}

/**
 * 获取学生选项列表
 */
export function getStudentOptions(keyword?: string) {
  return request({
    url: '/repair/students/options',
    method: 'get',
    params: { keyword }
  })
}

/**
 * 获取报修类型选项
 */
export function getRepairTypeOptions() {
  return [
    { label: '水电维修', value: 'water' },
    { label: '电路维修', value: 'electric' },
    { label: '门窗维修', value: 'door' },
    { label: '家具维修', value: 'furniture' },
    { label: '其他', value: 'other' }
  ]
}

/**
 * 获取报修状态选项
 */
export function getRepairStatusOptions() {
  return [
    { label: '待处理', value: 1, type: 'warning' },
    { label: '处理中', value: 2, type: 'primary' },
    { label: '已完成', value: 3, type: 'success' },
    { label: '已取消', value: 4, type: 'info' }
  ]
}

/**
 * 获取报修优先级选项
 */
export function getRepairPriorityOptions() {
  return [
    { label: '低', value: 1, type: 'info' },
    { label: '中', value: 2, type: 'warning' },
    { label: '高', value: 3, type: 'danger' }
  ]
}